gtars/src/uniwig/mod.rs

```diff
@@ -1,16 +1,722 @@
-//!
-//! # Uniwig BigWig generator
-
-//! The `uniwig` module is responsible for generating three BigWig output files based on a set of bed files. The generated files include a track for the start coordinates, a track for the end coordinates, and a track for the core coordinates.
-//!
-//! ## Under Construction
-//!
-//! This module is currently under construction. Stay tuned for more updates.
-//!
-//! ![Construction Sign](https://www.signoutfitters.com/images/products/detail/Workers_Symbol_Construction_Sign.png)
-//!
-//!
-
-pub fn run_uniwig() {
-    println!("Im running.")
+use clap::ArgMatches;
+use std::io::{BufRead, BufReader, Read, Write};
+use std::path::Path;
+use std::fs::{File, OpenOptions};
+use std::error::Error;
+use clap::builder::OsStr;
+use flate2::read::GzDecoder;
+use ndarray::Array;
+use ndarray_npy::write_npy;
+
+
+pub mod cli;
+
+pub mod consts {
+    pub const UNIWIG_CMD: &str = "uniwig";
+
 }
+
+pub struct Chromosome {
+    chrom: String,
+    starts: Vec<i32>,
+    ends: Vec<i32>,
+}
+impl Clone for Chromosome {
+    fn clone(&self) -> Self {
+        Self {
+            chrom: self.chrom.clone(), // Clone the string
+            starts: self.starts.clone(), // Clone the vector
+            ends: self.ends.clone(),   // Clone the vector
+        }
+    }
+}
+
+
+// pub fn read_bed_map(combinedbedpath: &str){
+//
+//
+// }
+
+pub fn read_bed_vec(combinedbedpath: &str) -> Vec<Chromosome> {
```

Comment:
What do you think about having this function return the `Result` enum? So instead, it will return:
```rs
Result<Vec<Chromosome>>
```
Since there are lot of issues that can arise with reading from files....

Reply:
Also, snake case for this variable? `combinedbedpath` --> `combined_bed_path`

Reply:
Also also, how about this:
```rs
pub fn read_bed_vec<P: AsRef<Path>>(combinedbedpath: P) -> Vec<Chromosome> { }
```
The trait bounds make it more flexible for what it can accept as input: https://stackoverflow.com/questions/62603012/rewriting-a-function-to-accept-asrefpath-instead-of-path